sdk/storage/inc/common/crypt.hpp

```diff
@@ -3,13 +3,65 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 namespace Azure { namespace Storage {
 
-  std::string Sha256(const std::string& text);
-  std::string Hmac_Sha256(const std::string& text, const std::string& key);
   std::string Base64Encode(const std::string& text);
   std::string Base64Decode(const std::string& text);
 
+  class Md5 {
+  public:
+    Md5();
+    ~Md5();
+
+    void Update(const uint8_t* data, std::size_t length);
+
+    std::string Digest() const;
+
+    static std::string Hash(const uint8_t* data, std::size_t length)
```

Comment:
A one-liner saying that this API returns Base64 encoded string would be great.

Reply:
It returns binary